sklearn/neighbors/binary_tree.pxi

```diff
@@ -1322,7 +1258,7 @@ cdef class BinaryTree:
 
         # flatten X, and save original shape information
         np_Xarr = X.reshape((-1, self.data.shape[1]))
-        cdef DTYPE_t[:, ::1] Xarr = get_memview_DTYPE_2D(np_Xarr)
+        cdef const DTYPE_t[:, ::1] Xarr = np_Xarr
```

Comment:
Input array can be read-only, in which case setting it to a non `const` memoryview would fail.